File: backend/src/Hotel.Orbital.Core/Models/LeisureDto.cs
using System.ComponentModel.DataAnnotations;
using Entities;

namespace Core.Models;

/// <summary>
/// Dto досуга
/// </summary>
public class LeisureDto
{
    /// <summary>
    /// Идентификатор
    /// </summary>
    [Required]
    public Guid Id { get; set; }
    /// <summary>
    /// Название
    /// </summary>
    [Required]
    public string Title { get; set; }
    
    /// <summary>
    /// Описание
    /// </summary>
    [Required]
    public string Description { get; set; }
    
    /// <summary>
    /// Заметка
    /// </summary>
    public string? Note { get; set; }
    
    /// <summary>
    /// Маршрут
    /// </summary>
    public string? Route { get; set; }
    
    /// <summary>
    /// Номер телефона
    /// </summary>
    [Phone]
    public string? PhoneNumber { get; set; }
    
    /// <summary>
    /// Электронная почта
    /// </summary>
    [EmailAddress]
    public string? Email { get; set; }
    
    /// <summary>
    /// Обложка
    /// </summary>
    [Required]
    public ImageDto Cover { get; set; }
    
    /// <summary>
    /// Изображения
    /// </summary>
    [Required]
    public List<ImageDto> Images { get; set; }

    /// <summary>
    /// Рассписание
    /// </summary>
    [Required]
    public List<LeisureDay> Days { get; set; }
    
    /// <summary>
    /// Дата и время создания
    /// </summary>
    [Required]
    public DateTimeOffset CreatedAt { get; set; }
}

File: web/src/features/leisures/api/dto/leisure-dto.js
import { mapFromImageDto } from "@/features/images";
import { mapFromLeisureDayDto } from "./leisure-day-dto";

export const mapFromLeisureDto = (dto) => ({
  id: dto.id,
  title: dto.title,
  description: dto.description,
  note: dto.note,
  route: dto.route,
  phone: dto.phoneNumber,
  email: dto.email,
  cover: mapFromImageDto(dto.cover),
  images: dto.images.map(mapFromImageDto),
  days: dto.days.map(mapFromLeisureDayDto),
  createdAt: dto.createdAt,
});
